pkg/urbit/vere/db/lmdb.c

```diff
@@ -1,627 +0,0 @@
-                  void*       ptr_v,
-                  const c3_c* key_c,
-                  void (*read_f)(void*, size_t, void*))
-{
-  MDB_txn* txn_u;
-  MDB_dbi  mdb_u;
-  c3_w     ret_w;
-
-  //  create a read transaction
-  //
-  if ( (ret_w = mdb_txn_begin(env_u, 0, MDB_RDONLY, &txn_u)) ) {
-    mdb_logerror(stderr, ret_w, "lmdb: meta read: txn_begin fail");
-    return read_f(ptr_v, 0, 0);
-  }
-
-  //  open the database in the transaction
-  //
-  if ( (ret_w =  mdb_dbi_open(txn_u, "META", 0, &mdb_u)) ) {
-    mdb_logerror(stderr, ret_w, "lmdb: meta read: dbi_open fail");
-    mdb_txn_abort(txn_u);
-    return read_f(ptr_v, 0, 0);
-  }
-
-  //  read by string key, invoking callback with result
-  {
-    MDB_val key_u = { .mv_size = strlen(key_c), .mv_data = (void*)key_c };
-    MDB_val val_u;
-
-    if ( (ret_w = mdb_get(txn_u, mdb_u, &key_u, &val_u)) ) {
-      mdb_logerror(stderr, ret_w, "lmdb: read failed");
-      mdb_txn_abort(txn_u);
-      return read_f(ptr_v, 0, 0);
-    }
-    else {
-      read_f(ptr_v, val_u.mv_size, val_u.mv_data);
-
-      //  read-only transactions are aborted when complete
-      //
-      mdb_txn_abort(txn_u);
-    }
-  }
-}
-
-/* u3_lmdb_save_meta(): save by string into the META db.
-*/
-c3_o
-u3_lmdb_save_meta(MDB_env*    env_u,
-                  const c3_c* key_c,
-                  size_t      val_i,
-                  void*       val_p)
-{
-  MDB_txn* txn_u;
-  MDB_dbi  mdb_u;
-  c3_w     ret_w;
-
-  //  create a write transaction
-  //
-  if ( (ret_w = mdb_txn_begin(env_u, 0, 0, &txn_u)) ) {
-    mdb_logerror(stderr, ret_w, "lmdb: meta write: txn_begin fail");
-    return c3n;
-  }
-
-  //  opens the database in the transaction
-  //
-  if ( (ret_w = mdb_dbi_open(txn_u, "META", MDB_CREATE, &mdb_u)) ) {
-    mdb_logerror(stderr, ret_w, "lmdb: meta write: dbi_open fail");
-    mdb_txn_abort(txn_u);
-    return c3n;
-  }
-
-  //  put value by string key
-  //
-  {
-    MDB_val key_u = { .mv_size = strlen(key_c), .mv_data = (void*)key_c };
-    MDB_val val_u = { .mv_size = val_i,         .mv_data = val_p };
-
-    if ( (ret_w = mdb_put(txn_u, mdb_u, &key_u, &val_u, 0)) ) {
-      mdb_logerror(stderr, ret_w, "lmdb: write failed");
-      mdb_txn_abort(txn_u);
-      return c3n;
-    }
-  }
-
-  //  commit txn
-  //
-  if ( (ret_w = mdb_txn_commit(txn_u)) ) {
-    mdb_logerror(stderr, ret_w, "lmdb: meta write: commit failed");
-    return c3n;
-  }
-
-  return c3y;
-}
-
-#if !defined(U3_OS_mingw)
-/* mdb_logerror(): writes an error message and lmdb error code to f.
-*/
-void mdb_logerror(FILE* f, int err, const char* fmt, ...)
-{
-  va_list ap;
-  va_start(ap, fmt);
-  vfprintf(f, fmt, ap);
-  va_end(ap);
-  fprintf(f, ": %s\r\n", mdb_strerror(err));
-}
-
-/* mdb_get_filesize(): gets the size of a lmdb database file on disk.
-*/
-intmax_t mdb_get_filesize(mdb_filehandle_t han_u)
-{
-  struct stat sat_u;
-  fstat(han_u, &sat_u);
-  return (intmax_t)sat_u.st_size;
-}
-#endif
```

Comment:
The mingw build is failing because of issues with `mdb_strerror()` -- you'll need to replicate this pattern with `mdb_logerror()`

Reply:
For context, see `compat/mingw/lmdb.patch`.